### PR TITLE
OMID-264 Fix deprecated WARNING in check-license stage

### PR DESCRIPTION
**Pull request by NihalJain**

### CODE REVIEW

**Comment by NihalJain** [comment]
Ensured warning is no longer shown during build.

Tested by first introducing as license error by deleting header from one of existing java file and then running mvn license:check both with/without change, failing as expected.

Next ran mvn license:format to fix header issue.

CC: @stoty

File: pom.xml [hunk 1]
```diff
@@ -367,54 +367,58 @@
...
-                        <exclude>**/pom.xml</exclude>
+                                <!-- Exclude pom -->
+                                <exclude>**/pom.xml</exclude>
 
-                        <!-- Exclude config and scripts -->
-                        <exclude>**/dev-utils/*</exclude>
-                        <exclude>dev-support/artifacts/**</exclude>
-                        <exclude>dev-support/work/**</exclude>
+                                <!-- Exclude config and scripts -->
+                                <exclude>**/dev-utils/*</exclude>
```
Comment (stoty) [inline pom.xml:395]: There is no such file in the repo.
Can we remove this ?

File: pom.xml [hunk 2]
```diff
@@ -367,54 +367,58 @@
...
+                                <exclude>dev-support/artifacts/**</exclude>
```
Comment (stoty) [inline pom.xml:396]: Can you check all of these ?
Looks like this was copied from HBase or Phoenix.
We should clean up the non-existent directories and files 
(except for the ones generated during build)

File: pom.xml [hunk 3]
```diff
@@ -367,54 +367,58 @@
...
-                        <include>src/**/*.proto</include>
-                        <include>**/*.css</include>
-                        <include>**/*.sh</include>
-                    </includes>
-                    <excludes>
-                        <!-- Add files to exclude from licensing -->
+                    <!-- Add comment style for protocol buffers -->
+                    <mapping>
+                        <java>SLASHSTAR_STYLE</java>
+                        <proto>DOUBLESLASH_STYLE</proto>
```
Comment (stoty) [inline pom.xml:375]: Just replace the ASL Header in the proto file with the standard Slashstar format one ?
That's what HBase and Phoenix uses.

File: pom.xml [hunk 4]
```diff
@@ -367,54 +367,58 @@
...
+                                <include>**/*.css</include>
+                                <include>**/*.sh</include>
+                            </includes>
+                            <excludes>
+                                <!-- Add files to exclude from licensing -->
 
-                        <!-- Exclude pom -->
-                        <exclude>**/pom.xml</exclude>
+                                <!-- Exclude pom -->
+                                <exclude>**/pom.xml</exclude>
```
Comment (stoty) [inline pom.xml:392]: POM files should have an ASL header,

File: pom.xml [hunk 5]
```diff
@@ -367,54 +367,58 @@
...
+                                <!-- Exclude assembly -->
+                                <exclude>**/maven/assembly/*</exclude>
 
-                        <!-- Project specific exclusions (e.g. they've been copied/modified from other open-source projects -->
+                                <!-- Project specific exclusions (e.g. they've been copied/modified from other open-source projects -->
 
-                        <!-- Protocol buffers generated classes -->
-                        <exclude>**/src/main/java/org/apache/omid/proto/TSOProto.java</exclude>
+                                <!-- Protocol buffers generated classes -->
+                                <exclude>**/src/main/java/org/apache/omid/proto/TSOProto.java</exclude>
```
Comment (stoty) [inline pom.xml:406]: No such file, we are generating this dynamically now.

File: pom.xml [hunk 6]
```diff
@@ -367,54 +367,58 @@
...
 
-                        <!-- Taken from https://github.com/brianfrankcooper/YCSB -->
-                        <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/Generator.java</exclude>
-                        <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/IntegerGenerator.java</exclude>
-                        <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/ZipfianGenerator.java</exclude>
-                        <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/ScrambledZipfianGenerator.java
-                        </exclude>
+                                <!-- Taken from https://github.com/brianfrankcooper/YCSB -->
+                                <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/Generator.java</exclude>
```
Comment (stoty) [inline pom.xml:409]: These should not be excluded.
Instead, the if the license is indeed ASL compatible (I think it is), then it should be added as a valid license.

File: pom.xml [hunk 7]
```diff
@@ -367,54 +367,58 @@
...
+                                <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/IntegerGenerator.java</exclude>
+                                <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/ZipfianGenerator.java</exclude>
+                                <exclude>**/src/main/java/org/apache/omid/benchmarks/utils/ScrambledZipfianGenerator.java
+                                </exclude>
 
 
-                        <!-- Taken from https://github.com/apache/hbase -->
-                        <exclude>**/src/main/java/org/apache/omid/committable/hbase/RegionSplitter.java</exclude>
+                                <!-- Taken from https://github.com/apache/hbase -->
+                                <exclude>**/src/main/java/org/apache/omid/committable/hbase/RegionSplitter.java</exclude>
```
Comment (stoty) [inline pom.xml:417]: This is already ASL.
I see no need to exclude it, just because it was copied.

File: pom.xml [hunk 8]
```diff
@@ -367,54 +367,58 @@
...
-                        <!-- Add files to exclude from licensing -->
+                    <!-- Add comment style for protocol buffers -->
+                    <mapping>
+                        <java>SLASHSTAR_STYLE</java>
+                        <proto>DOUBLESLASH_STYLE</proto>
+                    </mapping>
+                    <licenseSets>
+                        <licenseSet>
+                            <header>${LicenseSet.header}</header>
+                            <includes>
```
Comment (stoty) [inline pom.xml:380]: Maybe add Yaml files, and add ASL header there ?

**Review by stoty — CHANGES_REQUESTED** [review]

**Review by stoty — APPROVED** [review]
I realize that my comments are out of scope for this change.

Can you make a follow-up ticket to really clean up the configuration ?

**Comment by NihalJain** [comment]
> Can you make a follow-up ticket to really clean up the configuration ?

Sure let me clean up with a follow up JIRA. 

Also do you think we should replace this with spotless? I am well versed with it and based on what I am seeing this plugin is only for headers, which spotless does seemlessly. 

I am okay just cleaning up as well. Let me know what do you think.

**Comment by stoty** [comment]
I don't know spotless well.
Lets do whatever HBase is doing.

**Comment by NihalJain** [comment]
> I don't know spotless well.
> Lets do whatever HBase is doing.

HBase uses spotless. Let me create a jira to move to spotless and clean up.

**Pull request merged**